lib/Net/Async/Blockchain/Omni.pm

```diff
@@ -92,29 +92,48 @@ async sub transform_transaction {
 
     # the command listtransactions will guarantee that this transactions is from or to one
     # of the node addresses.
-    my $received_transaction;
-    my $detailed_transaction;
+    my ($omni_transaction, $parent_transaction);
+
     try {
-        $received_transaction = await $self->rpc_client->get_transaction($decoded_raw_transaction->{txid});
-        $detailed_transaction   = await $self->rpc_client->get_detailed_transaction($decoded_raw_transaction->{txid});
+        $omni_transaction   = await $self->rpc_client->get_transaction($decoded_raw_transaction->{txid});
+        $parent_transaction = await $self->rpc_client->get_detailed_transaction($decoded_raw_transaction->{txid});
     }
     catch {
         # transaction not found
         return undef;
     }
 
     # transaction not found, just ignore.
-    return undef unless $received_transaction && $received_transaction->{ismine};
+    return undef unless $omni_transaction && $omni_transaction->{ismine};
 
-    my $amount = Math::BigFloat->new($received_transaction->{amount});
-    my $fee    = Math::BigFloat->new($received_transaction->{fee} // 0);
-    my $block  = Math::BigInt->new($received_transaction->{block});
-    my ($from, $to) =
-        await Future->needs_all(map { $self->rpc_client->validate_address($received_transaction->{$_}) } qw(sendingaddress referenceaddress));
+    my @transaction = await _process_transaction($self, $omni_transaction, $parent_transaction);
 
-    my %category;
+    if (@transaction) {
```

Comment:
This `if` can be removed. When `@transaction` is empty, then the body of `for` will not be executed.